sci-libs/scotch/scotch-7.0.8.ebuild

```diff
@@ -0,0 +1,174 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs flag-o-matic multilib
+
+SOVER=$(ver_cut 1)
+
+DESCRIPTION="Software for graph, mesh and hypergraph partitioning"
+HOMEPAGE="https://www.labri.u-bordeaux.fr/perso/pelegrin/scotch/ https://gitlab.inria.fr/scotch/scotch"
+SRC_URI="https://gitlab.inria.fr/${PN}/${PN}/-/archive/v${PV}/${PN}-v${PV}.tar.bz2"
+S="${WORKDIR}/${PN}-v${PV}"
+
+LICENSE="CeCILL-2"
+SLOT="0/${SOVER}"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc int64 mpi static-libs test tools +threads"
+# bug #532620
+REQUIRED_USE="test? ( threads )"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	sys-libs/zlib
+	mpi? ( virtual/mpi )"
+RDEPEND="${DEPEND}"
+
+static_to_shared() {
+	local libstatic=${1}; shift
+	local libname=$(basename ${libstatic%.a})
+	local soname=${libname}$(get_libname ${SOVER})
+	local libdir=$(dirname ${libstatic})
+
+	einfo "Making ${soname} from ${libstatic}"
+	if [[ ${CHOST} == *-darwin* ]] ; then
+		${LINK:-$(tc-getCC)} ${LDFLAGS}  \
+			-dynamiclib -install_name "${EPREFIX}"/usr/lib/"${soname}" \
+			-Wl,-all_load -Wl,${libstatic} \
+			"$@" -o ${libdir}/${soname} || die "${soname} failed"
+	else
+		${LINK:-$(tc-getCC)} ${LDFLAGS}  \
+			-shared -Wl,-soname=${soname} \
+			-Wl,--whole-archive ${libstatic} -Wl,--no-whole-archive \
+			"$@" -o ${libdir}/${soname} || die "${soname} failed"
+		VERS_COMP=${PV//.}
+		[[ "${#VERS_COMP}" -gt 1 ]] && \
+			ln -s ${soname} ${libdir}/${libname}$(get_libname ${SOVER})
+		ln -s ${soname} ${libdir}/${libname}$(get_libname)
```

Comment:
Missing die.

Reply:
Adding the dies causes the install to fail now. 
```
 * Making libscotcherr.so.7 from lib/libscotcherr.a
ln: failed to create symbolic link 'lib/libscotcherr.so.7': File exists
```
Investigating...

Reply:
Removing the symlinking had no effect on the image and revdeps (`mumps`, `pastix`, `petsc`) installed fine.
```
$ sudo qa-cmp -I sci-libs/scotch
apn@APNsL13 ~/git/gentoo/gentoo/sci-libs/scotch $ diff -Naru scotch-7.0.8*
--- scotch-7.0.8.ebuild	2025-07-29 14:52:23.729672075 +0200
+++ scotch-7.0.8-r1.ebuild	2025-07-29 14:54:00.503081269 +0200
@@ -42,6 +42,8 @@
 			-shared -Wl,-soname=${soname} \
 			-Wl,--whole-archive ${libstatic} -Wl,--no-whole-archive \
 			"$@" -o ${libdir}/${soname} || die "${soname} failed"
+		[[ "${#VERS_COMP}" -gt 1 ]] && \
+			ln -s ${soname} ${libdir}/${libname}$(get_libname ${SOVER})
 		ln -s ${soname} ${libdir}/${libname}$(get_libname) || die
 	fi
 }
```